common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -139,14 +144,24 @@ public void run() {
                 Telemetry.getInstance().flush(correlationId);
                 EstsTelemetry.getInstance().flush(command, commandResult);
 
-                sExecutingCommands.remove(command);
-
                 //Return the result via the callback
-                returnCommandResult(command, commandResult, handler);
+                sExecutingCommandMap.remove(command).setResult(commandResult);
             }
         });
     }
 
+    private static BiConsumer<CommandResult, Throwable> getCommandResultConsumer(
+            @NonNull final BaseCommand command,
+            @NonNull final Handler handler) {
+        return new BiConsumer<CommandResult, Throwable>() {
+            @Override
+            public void accept(CommandResult result, Throwable throwable) {
+                // We can ignore the second (exception) param, not used.
```

Comment:
Well, do we?  Long term, it's a better way of communicating error states.  The contract here is that either you have a result or you have a throwable.  Mostly, the code for these looks like if (t == null) { error handling; return; } result handling;

Reply:
Applied in 91ed0a71841cdf4095b7fac398dda2330df16da7